### PR TITLE
fix: open url err invalid argument type on windows WSL systems

### PR DESCRIPTION
**Pull request by kchindam-infy**

### CODE REVIEW

File: lib/utils/open-url.js [hunk 1]
```diff
@@ -39,10 +42,29 @@ const openUrl = async (npm, url, title, isFile) => {
     assertValidUrl(url)
   }
 
+  const isWSL = async () => {
+    try {
+      const { stdout } = await exec('uname -r')
```
Comment (wraithgar) [inline lib/utils/open-url.js:47]: This is not gonna work, we don't call bare `exec` to spawn things from npm.  This also seems like a pretty heavy-handed way to detect this.

Comment (wraithgar) [reply lib/utils/open-url.js:47]: This kind of logic should live in `@npmcli/promise-spawn` in any case, that's its whole reason for existing.

Comment (wraithgar) [reply lib/utils/open-url.js:47]: It looks like we're already detecting wsl in `promise-spawn` https://github.com/npm/promise-spawn/blob/7928731de498e4dcdfab60ecb1d49040fec521a7/lib/index.js#L133-L137

**Comment by kchindam-infy** [comment]
After further deeper analysis the issue appears to be with Node and how the WSL environment is handled child_spawn process. The npm cli promise spawn library is treating WSL as win32 and the shell configuration is calculated accordingly.

**Pull request closed without merging**